### PR TITLE
Feature/cu 86dudyd44/backend connectors and matrix mapping

### PR DESCRIPTION
**Pull request by rainierlightit**
# ⚡ {0} - [{1}](https://app.clickup.com/t/{2}) ⚡

## 💻 What type of change is this?

{3}

## ⭐ Description

For more background, see ticket **{1}[in-progress].**

### Before

image

### After

image

### To review
- [ ] I have tested this change locally in multiple screen sizes
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] If my task include an endpoint, I add the endpoint to Hopscotch/Postman Project
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Backoffice/Users/App/Integrations/pVerify/Requests/EligibilitySummary.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Lightit\Backoffice\Users\App\Integrations\pVerify\Requests;
+
+use Lightit\Backoffice\Users\Domain\DataTransferObjects\pVerifyEligibilitySummaryRequestDTO;
+use Saloon\Http\Request;
+
+class EligibilitySummary extends Request
+{
+    public function __construct(
+        protected pVerifyEligibilitySummaryRequestDTO $pVerifyEligibilitySummary,
```
Comment (facundodartayete) [inline src/Backoffice/Users/App/Integrations/pVerify/Requests/EligibilitySummary.php:13]: Casing in the class

File: src/Backoffice/Users/App/Integrations/pVerify/Requests/EligibilitySummary.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    ) {
+    }
+
+    public function resolveEndpoint(): string
+    {
+        return '/EligibilitySummary';
+    }
+
+    public function body(): array
```
Comment (facundodartayete) [inline src/Backoffice/Users/App/Integrations/pVerify/Requests/EligibilitySummary.php:22]: should we have a `createDTOFromResponse()` or we don't know how that would look?

File: src/Backoffice/Users/App/Integrations/pVerify/Requests/GetToken.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Backoffice\Users\App\Integrations\pVerify\Requests;
+
+use Lightit\Backoffice\Users\App\Integrations\pVerify\pVerifyConnector;
```
Comment (facundodartayete) [inline src/Backoffice/Users/App/Integrations/pVerify/Requests/GetToken.php:7]: name casing

File: src/Backoffice/Users/Domain/DataTransferObjects/pVerifyEligibilitySummaryRequestDTO.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        public string $referenceId,
+        public string $Location,
+        public string $PracticeTypeCode,
+        public string $IncludeTextResponse,
+        public string $InternalId,
+        public string $CustomerID,
+    ) {
+    }
+
+    public function toArray(): array
```
Comment (facundodartayete) [inline src/Backoffice/Users/Domain/DataTransferObjects/pVerifyEligibilitySummaryRequestDTO.php:29]: what do we need this for?

File: src/Shared/App/Console/Commands/ProcessQuestMatrix.php [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+use Lightit\Backoffice\DMEProviders\Domain\Models\DMEProvider;
+use Lightit\Backoffice\Payers\Domain\Actions\BulkCreatePayers;
+use Lightit\Backoffice\Payers\Domain\DataTransferObjects\PayerDTO;
+use Lightit\Backoffice\Payers\Domain\Enums\PayerType;
+use Lightit\Backoffice\Payers\Domain\Models\Payer;
+use Lightit\Shared\Domain\Actions\BulkCreatePayersDMEProviders;
+use Lightit\Shared\Domain\Actions\GetQuestMatrix;
+use Lightit\Shared\Domain\DataTransferObjects\PayersDMEProvidersDTO;
+
+class ProcessQuestMatrix extends Command
```
Comment (facundodartayete) [inline src/Shared/App/Console/Commands/ProcessQuestMatrix.php:24]: Command classes are named with the `Command` suffix

File: src/Shared/App/Console/Commands/ProcessQuestMatrix.php [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+        private readonly BulkCreatePayers $bulkCreatePayers,
+        private readonly BulkCreateDMEProviders $bulkCreateDMEProviders,
+        private readonly BulkCreatePayersDMEProviders $bulkCreatePayersDMEProviders,
+        private readonly GetQuestMatrix $getQuestMatrix,
+    ) {
+        parent::__construct();
+    }
+
+    public function handle(
+
```
Comment (facundodartayete) [inline src/Shared/App/Console/Commands/ProcessQuestMatrix.php:47]: nit: unnecessary line

File: src/Shared/App/Console/Commands/ProcessQuestMatrix.php [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+    /**
+     * @param Collection<string, Collection<int|string, mixed>> $matrixData
+     *
+     */
+    private function generateDMEProviders(Collection $matrixData, Datasync $datasync): int
+    {
+        $dmeProviders = $matrixData->select('ADC_DME_Name__c')->unique();
+        $fullDmeData = collect();
+        foreach ($dmeProviders as $dmeProvider) {
```
Comment (facundodartayete) [inline src/Shared/App/Console/Commands/ProcessQuestMatrix.php:97]: nit: could also be a map iteration

**Review by facundodartayete — APPROVED** [review]

File: src/Backoffice/Users/App/Integrations/pVerify/Requests/EligibilitySummary.php (hunk 1 above)
Comment (rainierlightit) [reply src/Backoffice/Users/App/Integrations/pVerify/Requests/EligibilitySummary.php:13]: should we use capital P? I understand the name of the company is written with an lowercase p for a reason, that's why I didn't capitalize it.

File: src/Backoffice/Users/App/Integrations/pVerify/Requests/EligibilitySummary.php (hunk 2 above)
Comment (rainierlightit) [reply src/Backoffice/Users/App/Integrations/pVerify/Requests/EligibilitySummary.php:22]: After discussing that technique with Sergio, the conclussion was that those createFrom methods should not be used unless we have different data mapings for a DTO creation, in which case we should use a factory class.

File: src/Backoffice/Users/App/Integrations/pVerify/Requests/GetToken.php (hunk 3 above)
Comment (rainierlightit) [reply src/Backoffice/Users/App/Integrations/pVerify/Requests/GetToken.php:7]: same as the comment above.

File: src/Backoffice/Users/Domain/DataTransferObjects/pVerifyEligibilitySummaryRequestDTO.php (hunk 4 above)
Comment (rainierlightit) [reply src/Backoffice/Users/Domain/DataTransferObjects/pVerifyEligibilitySummaryRequestDTO.php:29]: to build te request body in the saloon request class once we know what should go in the body.

File: src/Backoffice/Users/App/Integrations/pVerify/Requests/EligibilitySummary.php (hunk 1 above)
Comment (facundodartayete) [reply src/Backoffice/Users/App/Integrations/pVerify/Requests/EligibilitySummary.php:13]: but it is a convention, classes should always start with uppercase

**Pull request merged**